### PR TITLE
add count missing annotations

### PR DESCRIPTION
**Pull request by ablack12**

### CODE REVIEW

File: migrations/interface.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ func (m *migrationRegistry) registerMigration(name string, factory MigrationFact
 func (m *migrationRegistry) Migration(name string, opts MigrationOptions) (Migration, error) {
 	migrationFactory, ok := m.migrations[name]
 	if !ok {
-		return nil, errors.Errorf("no migration exists for name '%s'", name)
+		return nil, errors.Errorf("no migration exists for ttlMigrationName '%s'", name)
```
Comment (ybrill) [inline migrations/interface.go]: I'm not sure I get these changes. This function and `validate` would be called for _any_ migration, not just the TTL one.

File: migrations/countMissingAnnotations.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+func (c *CountMissingAnnotations) Execute(ctx context.Context, client *mongo.Client) error {
+	query := bson.M{
+		"$match": bson.M{
+			task.ProjectKey:   "mongodb-mongo-v8.0",
+			task.RequesterKey: evergreen.RepotrackerVersionRequester,
+			task.StatusKey:    evergreen.TaskFailed,
+			bsonutil.GetDottedKeyName(task.DetailsKey, task.TaskEndDetailType): evergreen.CommandTypeTest,
+			task.HasAnnotationsKey: false,
+			task.CreateTimeKey:     bson.M{"$gte": "2024-05-24T00:00:00.000Z"},
```
Comment (ybrill) [inline migrations/countMissingAnnotations.go]: Does this work to check a time? I would have thought we'd need to compare against a go `time.Time`.

**Review by ybrill — COMMENTED** [review]
it seems like we might have trouble running this in Argo (and/or anywhere in k8s), given how running jobs are liable to get interrupted.

File: migrations/interface.go (hunk 1 above)
Comment (ablack12) [reply migrations/interface.go]: oooops thanks for catching, this was just that i did an editor-driven refactor and it must've been more aggressive than i expected 🙃

File: migrations/countMissingAnnotations.go (hunk 2 above)
Comment (ablack12) [reply migrations/countMissingAnnotations.go]: Maybe it does; I can switch it just to be sure.

File: migrations/ttlCollection_test.go [hunk 3]
```diff
@@ -136,8 +136,8 @@ func TestNewTTLCollection(t *testing.T) {
 		t.Setenv(ttlFieldEnvVar, "tasks")
 		_, err := NewTTLCollection(MigrationOptions{})
 		assert.Error(t, err)
-		assert.Contains(t, err.Error(), "database name not specified")
-		assert.Contains(t, err.Error(), "collection name not specified")
+		assert.Contains(t, err.Error(), "database ttlMigrationName not specified")
+		assert.Contains(t, err.Error(), "collection ttlMigrationName not specified")
```
Comment (ybrill) [inline migrations/ttlCollection_test.go]: I think these will be unchanged

**Review by ybrill — COMMENTED** [review]
I wonder if, for read-only operations like this, we'd want to add an option to run with a secondary read-preference.

File: migrations/ttlCollection_test.go (hunk 3 above)
Comment (ablack12) [reply migrations/ttlCollection_test.go]: dang i thought i found all of these my bad

**Comment by ablack12** [comment]
I think this search shouldn't be that expensive so i'm inclined to just try it out but we can see!

File: migrations/countMissingAnnotations.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+			task.CreateTimeKey:     bson.M{"$gte": timeToCheck},
+		},
+	}
+
+	cursor, err := client.Database(c.database).Collection(task.Collection).Find(ctx, query)
+	if err != nil {
+		return errors.Wrap(err, "aggregating tasks")
+	}
+	tasksToCheck := []task.Task{}
+	err = cursor.All(ctx, &tasksToCheck)
```
Comment (ybrill) [inline migrations/countMissingAnnotations.go:64]: If it's going to be a lot of tasks maybe we'd want to iterate the cursor one-by-one so as not to have them all in memory at once?

**Review by ybrill — APPROVED** [review]

**Pull request merged**